packages/doenetml-worker-rust/lib-doenetml-core/src/core/props/data_query/filters.test.rs

```diff
@@ -0,0 +1,433 @@
+use std::rc::Rc;
+
+use super::super::*;
+use super::*;
+use crate::{test_utils::*, Core};
+
+#[test]
+fn can_apply_test_on_an_op() {
+    let op = Op::And(true, false);
+    assert_eq!(op.apply_test(&true), false);
+
+    let op = Op::And(true, true);
+    assert_eq!(op.apply_test(&true), true);
+
+    let op = Op::Or(true, false);
+    assert_eq!(op.apply_test(&true), true);
+
+    let op = Op::Or(OpNot(Op::And(true, false)), false);
+    assert_eq!(op.apply_test(&true), true);
+}
```

Comment:
If you make `bool` be the one that is implemented only for `GraphNode`, then you can also add this test:
```
    let op = OpNot(true);
    assert_eq!(op.apply_test(&true), false);
```